File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
           vector<int> r;
    int n = numbers.size();
    int i = 0;
    int j = n -1 ;
    while(i <= j){
        if(numbers[i] + numbers[j] == target){
            r.push_back(i+1);
            r.push_back(j+1);
            return r;
        } else if(numbers[i] + numbers[j] < target){
            i++;
        } else {
            j--;
        }
    }
    }
};

File: Data Structures & Algorithms/car-fleet/submission-0.py
class Solution:
    def carFleet(self, target: int, position: List[int], speed: List[int]) -> int:
        cars = []
        for pos, spd in zip(position, speed):
            cars.append((pos, spd))
        
        cars.sort(reverse=True)
        stack = []
        for car in cars:
            dist = target - car[0]
            time = dist / car[1]
            if len(stack) == 0:
                stack.append(time)
            else:
                if time > stack[-1]:
                    stack.append(time)
        
        return len(stack)